pkg/controller/proposal/controller.go

```diff
@@ -509,6 +513,22 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
 	}
 }
 
+func applyChangeToConfig(values map[string]*configapi.PathValue, path string, value *configapi.PathValue) (string, *configapi.PathValue) {
+	values[path] = value
```

Comment:
What’s the purpose of setting this key here?

Reply:
The core of the apply change function was to update the map (as was done before, but outside this function) and then to process the path parents to clear-out any prior deletions.